core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -310,9 +316,12 @@ public synchronized IndexShard createShard(ShardRouting routing) throws IOExcept
             eventListener.indexShardStateChanged(indexShard, null, indexShard.state(), "shard created");
             eventListener.afterIndexShardCreated(indexShard);
             indexShard.updateRoutingEntry(routing, true);
-            if (shards.isEmpty() && this.indexSettings.getTranslogSyncInterval().millis() != 0) {
+            if (shards.isEmpty()) {
                 ThreadPool threadPool = nodeServicesProvider.getThreadPool();
-                new AsyncTranslogFSync(this, threadPool).schedule(); // kick this off if we are the first shard in this service.
+                if (this.indexSettings.getTranslogSyncInterval().millis() != 0) {
+                    new AsyncTranslogFSync(this, threadPool); // kick this off if we are the first shard in this service.
```

Comment:
Move this comment up just inside the `if (shards.isEmpty()) {` saying "// kick off async ops for the first shard in this index"?